### DISCUSSION TITLE
Migration from BitBucket to GitHub Enterprise Cloud

### DISCUSSION THREAD

**Discussion by pankajmera** [community/community#54254] [Enterprise]
### Select Topic Area

Question

### Body

Hi,
I'm working on a project where I have a need of Migration from BitBucket code repository to GitHub Enterprise Cloud using the automation tool or automation scripts. 

I will have lots of repo in my existing repo which is BitBucket and needs to automate them to move onto GitHub Enterprise cloud.

The Script is less preferable but if that is the only choice then please share the details/steps.

**Comment by ghostinhershell** [comment]
Hi @pankajmera, 

You can use GitHub Enterprise Importer to migrate repositories from Bitbucket Server to GitHub Enterprise Cloud. Migrations from Bitbucket Server are only supported for Bitbucket Server or Bitbucket Data Center version 5.14+ or higher. Learn more about that here: https://docs.github.com/en/migrations/using-github-enterprise-importer/migrating-from-bitbucket-server-to-github-enterprise-cloud/about-migrations-from-bitbucket-server-to-github-enterprise-cloud. 

Please note that, at this time, Bitbucket _Cloud_ is not included in the GitHub Enterprise Importer offerings. We do have some recommendations for migrating data stored in Bitbucket Cloud. By utilizing our process for [Importing an external Git repository using the command line](https://docs.github.com/migrations/importing-source-code/using-the-command-line-to-import-source-code/importing-an-external-git-repository-using-the-command-line#title-h1), you should be able to migrate data stored in Bitbucket Cloud to a your GitHub repository.

If you don't wish to import utilizing the command line, we have also located some [documentation](https://confluence.atlassian.com/confeval/development-tools-evaluator-resources/bitbucket/bitbucket-migrate-bitbucket-cloud-to-bitbucket-server) which details a process for migrating the Bitbucket data into a Bitbucket Server first, which should then allow you to migrate from Server into a GitHub repository utilizing the Enterprise Importer. **It should also be noted that this particular process is outside of our support offerings and would be utilized at your own risk.**

As for the addition of Bitbucket Cloud to the GEI, we can't say if or when we may add a feature, but I recommend you keep an eye on the [GitHub Blog](https://github.com/blog) and our social media feeds for the latest announcements about new features and fixes. You can also browse our [Public roadmap](https://github.com/orgs/github/projects/4247/views/1) which is a project board that provides a high level overview of the current developments and their planned release schedule

Let me know if you have any additional questions. Thanks!

**Comment by pankajmera** [comment]
Thanks for responding, let me look at the details provided by you and get
back to you if needed.
